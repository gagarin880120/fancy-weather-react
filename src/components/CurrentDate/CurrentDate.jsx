import React from 'react';
import styles from './CurrentDate.module.css';
import PropTypes from 'prop-types';

export default function CurrentDate({ currentDate }) {
  return (
    <div className={styles.wrapper}>
      {currentDate}
    </div>
  );
}

CurrentDate.propTypes = {
  currentDate: PropTypes.string,
};

CurrentDate.defaultProps = {
  currentDate: '',
};
