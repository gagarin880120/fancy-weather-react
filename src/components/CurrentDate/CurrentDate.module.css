.wrapper {
  text-transform: capitalize;
  font-size: 20px;
  margin-top: 10px;
}
