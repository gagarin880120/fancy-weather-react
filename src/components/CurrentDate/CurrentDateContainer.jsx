import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import CurrentDate from './CurrentDate';
import { selectCurrentDate } from '../../redux/selectors';

export function CurrentDateContainer({ currentDate }) {
  return (
    <CurrentDate
      currentDate={currentDate}
    />
  );
}

export const mapStateToProps = (state) => ({
  currentDate: selectCurrentDate(state),
});

CurrentDateContainer.propTypes = {
  currentDate: PropTypes.string,
};

CurrentDateContainer.defaultProps = {
  currentDate: '',
};

export default connect(mapStateToProps)(CurrentDateContainer);
