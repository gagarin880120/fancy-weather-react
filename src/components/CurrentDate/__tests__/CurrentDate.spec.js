import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import CurrentDate from '../CurrentDate';

describe('CurrentDate', () => {
  describe('component', () => {
    test('should render without crashing', () => {
      const renderer = new ShallowRenderer();
      renderer.render(<CurrentDate />);

      const result = renderer.getRenderOutput();

      expect(result).toMatchSnapshot();
    });
  });
});
