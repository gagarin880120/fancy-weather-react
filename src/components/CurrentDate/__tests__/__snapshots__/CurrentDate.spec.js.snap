// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentDate component should render without crashing 1`] = `
<div
  className="wrapper"
>
  
</div>
`;
